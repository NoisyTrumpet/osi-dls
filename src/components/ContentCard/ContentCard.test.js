import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { ContentCard } from "./";

const exampleCard = {
  title: {
    title1: "title 1"
  },
  copy: {
    copy1: "copy 1",
    copy2: "copy 2",
    copy3: "copy 3"
  },
  href: {
    url: "https://osilife.com",
    target: "_blank",
    rel: ""
  },
  image: {
    alt: "Hallmark Logo",
    size: {
      mobile: "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg",
      tablet: "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg",
      desktop: "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg"
    }
  }
};

describe("ContentCard", () => {
  test("Should render", () => {
    const wrapper = shallow(
      <ContentCard type="media" theme="vertical" cardData={exampleCard} />
    );
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
