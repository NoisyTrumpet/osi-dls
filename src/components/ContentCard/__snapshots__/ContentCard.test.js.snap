// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard Should render 1`] = `
<div
  className="content-card vertical media-card"
  data-theme="vertical"
  data-type="media"
>
  <a
    className="card-container"
    href="https://www.hallmark.com"
    rel=""
    target="_blank"
  >
    <div
      aria-hidden="true"
      className="card-icon"
    />
    <Picture
      cardImage={
        Object {
          "alt": "Hallmark Logo",
          "size": Object {
            "desktop": "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg",
            "mobile": "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg",
            "tablet": "https://ideas.hallmark.com/wp-content/uploads/2020/06/2.jpg",
          },
        }
      }
    />
    <div
      className="card-content"
    >
      <h2
        className="card-title"
      >
        <Typography
          variant="eyebrow"
        >
          title 1
        </Typography>
      </h2>
      <p
        className="card-copy"
      >
        <Typography
          variant="body"
        >
          copy 1
        </Typography>
        <Typography
          variant="body"
        >
          copy 2
        </Typography>
        <Typography
          variant="body"
        >
          copy 3
        </Typography>
      </p>
    </div>
  </a>
</div>
`;
